Formula/elixir-ls.rb

```diff
@@ -0,0 +1,51 @@
+class ElixirLs < Formula
+  desc "Language Server Protocol implementation for Elixir"
+  homepage "https://github.com/JakeBecker/elixir-ls"
+  url "https://github.com/JakeBecker/elixir-ls/archive/v0.2.25.tar.gz"
+  sha256 "48aef4a0e795627962bde7e9c5a4fd201f92ef816fc823738d6a1702ee911ee0"
+
+  depends_on "elixir"
+
+  def install
+    # Ensure Hex and Rebar3 are installed
+    system "mix", "local.hex", "--force"
```

Comment:
Will this fetch a dependency? Cause in this case it will need to be versioned.

Reply:
There is a way to install a specifix version of Hex.

```sh
mix hex.install 0.20.1
```

`0.20.1` is the current version at the time of writing.

Should I just use `"0.20.1"` as the argument to `system`, or should the version for Hex be specified somewhere else?

Reply:
I have no idea how mix works. But formula aren't allowed to download unversioned things. https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things

Reply:
> Note that we now allow tools like cargo, gem and pip to download specifically versioned libraries during installation.

Can we treat `mix` as the tool of this category, and `hex` as a versioned dependency?

So, those are equivalent in Elixir and Ruby:

```sh
mix hex.install 0.20.1
gem install bundler -v 2.0.1
```

Reply:
That sounds reasonable. Is there no way to have a a lockfile or something for mix?

Reply:
Sadly, the lockfile (`mix.lock`) only concerns the dependencies of the application. Both `hex` and `rebar3` are package managers, and `mix local.{hex,rebar}` are the primary ways to install them.

I found the URL for rebar3, but I'm having troubles with Hex. I think that installing a direct version at build time is the best option. I'll make an update.